extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -576,12 +587,16 @@ private Optional<ResultSetInformation> getSampleResults(
       );
       return Optional.of(new ResultSetInformation(
           null,
-          // since the rows can be sampled, get the number of rows from counters
-          rowsAndSize.orElse(new Pair<>(null, null)).lhs,
-          rowsAndSize.orElse(new Pair<>(null, null)).rhs,
           dataSource,
           // only populate sample results in case a select query is successful
-          isSelectQuery ? SqlStatementResourceHelper.getResults(payload).orElse(null) : null
+          isSelectQuery ? SqlStatementResourceHelper.getResults(payload).orElse(null) : null,
+          ImmutableList.of(
+              new PageInformation(
+                  rowsAndSize.orElse(new Pair<>(null, null)).lhs,
```

Comment:
This should be populate from the counters. 
We need to rework 
```
SqlStatementResourceHelper.getRowsAndSizeFromPayload(
          payload,
          isSelectQuery
      );
      
```
so its returns pageInformation. 
The next set of changes that I am making then would not touch this piece of code.

Reply:
This might need to be handled at least partially in the next change, as counters don't really mean anything for results if the destination is the task report.